core/parsigneddata.go

```diff
@@ -0,0 +1,238 @@
+// Copyright © 2022 Obol Labs Inc.
+//
+// This program is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation, either version 3 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of  MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package core
+
+import (
+	"encoding/json"
+
+	"github.com/attestantio/go-eth2-client/spec"
+	eth2p0 "github.com/attestantio/go-eth2-client/spec/phase0"
+
+	"github.com/obolnetwork/charon/app/errors"
+)
+
+var (
+	_ ParSignedData = VersionedSignedBeaconBlock{}
+	_ ParSignedData = Attestation{}
+	_ ParSignedData = ParSig{}
+	_ ParSignedData = SignedExit{}
+)
+
+func NewParSig(sig eth2p0.BLSSignature, shareIdx int) ParSig {
+	return ParSig{BLSSignature: sig, shareIdx: shareIdx}
+}
+
+// ParSig is a partial signature that implements ParSignedData without additional data.
+type ParSig struct {
+	eth2p0.BLSSignature
+	shareIdx int
+}
+
+func (ParSig) DataRoot() (eth2p0.Root, error) {
+	// parsig (and randao) is just a signature, it doesn't have other data.
+	return eth2p0.Root{}, nil
+}
+
+func (r ParSig) Signature() Signature {
+	return SigFromETH2(r.BLSSignature)
+}
+
+func (r ParSig) ShareIdx() int {
+	return r.shareIdx
+}
+
+func (ParSig) MarshalData() ([]byte, error) {
```

Comment:
I suppose this is TBD

Reply:
No, ParSig doesn't have any data, it is only a signature, so this returns nil.